compiler/noirc_evaluator/src/brillig/brillig_gen/brillig_block.rs

```diff
@@ -798,23 +818,27 @@ impl<'block> BrilligBlock<'block> {
 
     /// Array set operation in SSA returns a new array or slice that is a copy of the parameter array or slice
     /// With a specific value changed.
+    ///
+    /// Skip setting a value and store the resulting length in the given SingleAddrVariable if
+    /// provided
     fn convert_ssa_array_set(
         &mut self,
         source_variable: BrilligVariable,
         destination_variable: BrilligVariable,
-        index_register: MemoryAddress,
-        value_variable: BrilligVariable,
+        index_and_value_or_len: Result<(MemoryAddress, BrilligVariable), SingleAddrVariable>,
```

Comment:
Are you doing this to use a `Result` like an either expression? I would prefer we don't do this (we can use enums or options).